R/build_xgboost.R

```diff
@@ -1380,6 +1380,21 @@ tidy.xgboost_exp <- function(x, type = "importance", pretty.name = FALSE, binary
         ret
       }
     },
+    evaluation_by_class = { # We assume this is called only for multiclass classification.
+      # Delegate showing error for failed models to grance().
+      if ("error" %in% class(x)) {
+        return(glance(x, pretty.name = pretty.name, ...))
+      }
+      # get evaluation scores from training data
+      actual <- extract_actual(x)
+      predicted <- extract_predicted_multiclass_labels(x)
+
+      per_level <- function(class) {
```

Comment:
It maybe better not to use "class" for the variable name since it is a common function.

Reply:
I'm now using the name "level" instead.